pkg/controllers/cnset/controller.go

```diff
@@ -124,6 +124,10 @@ func (c *Actor) Observe(ctx *recon.Context[*v1alpha1.CNSet]) (recon.Action[*v1al
 		return c.with(sts, svc).Scale, nil
 	}
 
+	err = v1alpha1.AddBucketFinalizer(ctx.Context, ctx.Client, cn.Deps.LogSet.ObjectMeta, v1alpha1.BucketCNFinalizer)
```

Comment:
A cluster may have multiple CN/DNSet attached, each set should add their unique finalizer (e.g. add their namespace and name to the finalizer)